recipe/build.sh

```diff
@@ -110,6 +115,41 @@ export TF_CONFIGURE_IOS=0
 sed -i -e "/PROTOBUF_INCLUDE_PATH/c\ " .bazelrc
 sed -i -e "/PREFIX/c\ " .bazelrc
 
+if [[ ${cuda_compiler_version} != "None" ]]; then
+
+    export CUDA_TOOLKIT_PATH=/usr/local/cuda-${cuda_compiler_version}
+    export TF_CUDA_PATHS="${PREFIX},/usr/local/cuda-${cuda_compiler_version},/usr"
+    export USE_CUDA=1
+    export cuda=Y
+    if [[ ${cuda_compiler_version} == 9.0* ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2
+    elif [[ ${cuda_compiler_version} == 9.2* ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0
+    elif [[ ${cuda_compiler_version} == 10.* ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0,7.2,7.5
+    elif [[ ${cuda_compiler_version} == 11.0* ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0,7.2,7.5,8.0
+    elif [[ ${cuda_compiler_version} == 11.1 ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0,7.2,7.5,8.0,8.6
+    elif [[ ${cuda_compiler_version} == 11.2 ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0,7.2,7.5,8.0,8.6
+    else
+        echo "unsupported cuda version. edit build_pytorch.sh"
+        exit 1
+    fi
+    
+    export TF_NEED_CUDA=1
+    export TF_CUDA_VERSION="${cuda_compiler_version}"
+    export TF_CUDNN_VERSION="${cudnn}"
+    export TF_CUDA_CLANG=0
+    export TF_DOWNLOAD_CLANG=0
+    export TF_NEED_TENSORRT=0
+    export TF_NCCL_VERSION=""
+    BUILD_OPTS="${BUILD_OPTS} --config=cuda --linkopt=-L${PREFIX}/lib --define=LIBDIR=${PREFIX}/lib"
```

Comment:
```suggestion
    BUILD_OPTS="${BUILD_OPTS} -s --config=cuda --linkopt=-L${PREFIX}/lib --define=LIBDIR=${PREFIX}/lib"
```

Apparently `-s` makes it print out the compilation commands.

Reply:
you might even want to try changing `--linkopt` to `--copt` or even `--host_linkopt`

Reply:
Thanks, yeah that is very verbose, but didn't really give any more information about what does wrong.

Reply:
how long does it take to get to failure?

Reply:
Time to failure varies, because the order isn't always the same. Can be anywhere from 30 minutes to several hours.

Reply:
sigh, this makes it harder to debug.

Reply:
> you might even want to try changing `--linkopt` to `--copt` or even `--host_linkopt`

I think `--copt=-L${PREFIX}/lib` might have done the trick! I'm almost afraid to get my hopes up at this point, but we are safely past the place where the build failed last time. Fingers crossed!

Reply:
Nope, it fails later with

```
[12,897 / 29,765] Compiling tensorflow/lite/toco/graph_transformations/resolve_multiply_by_zero.cc [for host]; 3s local ... (4 actions running)
ERROR: /home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/work/tensorflow/tools/proto_text/BUILD:31:10: Linking of rule '//tensorflow/tools/proto_text:gen_proto_text_functions' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command 
  (cd /home/conda/.cache/bazel/_bazel_conda/fc5b6f8a245ea6c7bfa068d002f44f78/execroot/org_tensorflow && \
  exec env - \
    PATH=/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/work:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_build_env/bin:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin:/opt/conda/condabin:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_build_env:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_build_env/bin:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin:/opt/conda/bin:/opt/conda/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/conda/bin:/usr/local/cuda/bin \
    PWD=/proc/self/cwd \
  external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc @bazel-out/host/bin/tensorflow/tools/proto_text/gen_proto_text_functions-2.params)
Execution platform: @local_execution_config_platform//:platform
/usr/bin/ld: cannot find -lprotobuf
/usr/bin/ld: cannot find -lsnappy
/usr/bin/ld: cannot find -lprotobuf
collect2: error: ld returned 1 exit status
ERROR: /home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/work/tensorflow/core/framework/BUILD:1261:31 Linking of rule '//tensorflow/tools/proto_text:gen_proto_text_functions' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command 
  (cd /home/conda/.cache/bazel/_bazel_conda/fc5b6f8a245ea6c7bfa068d002f44f78/execroot/org_tensorflow && \
  exec env - \
    PATH=/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/work:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_build_env/bin:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin:/opt/conda/condabin:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_build_env:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_build_env/bin:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac:/home/conda/feedstock_root/build_artifacts/tensorflow-split_1622642338317/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin:/opt/conda/bin:/opt/conda/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/conda/bin:/usr/local/cuda/bin \
    PWD=/proc/self/cwd \
  external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc @bazel-out/host/bin/tensorflow/tools/proto_text/gen_proto_text_functions-2.params)
Execution platform: @local_execution_config_platform//:platform
INFO: Elapsed time: 13126.052s, Critical Path: 444.51s
INFO: 13232 processes: 8181 internal, 5051 local.
FAILED: Build did NOT complete successfully
```

full log at https://gist.github.com/izahn/42712a8437361cb40aa5aed129f05155